cli/test/org/partiql/cli/functions/ReadFileTest.kt

```diff
@@ -151,4 +151,64 @@ class ReadFileTest {
 
         assertEquals(ion.singleValue(expected), actual)
     }
+
+    @Test
+    fun readExcelCsvFile() {
+        writeFile("simple_excel.csv", "title,category,price\nharry potter,book,7.99")
+
+        val args = listOf("\"${dirPath("simple_excel.csv")}\"", "{type:\"excel_csv\", header:true}").map { it.exprValue() }
+
+        val actual = function.call(env, args).ionValue
+        val expected = "[{title:\"harry potter\",category:\"book\",price:\"7.99\"}]"
+
+        assertEquals(ion.singleValue(expected), actual)
+    }
+
+    @Test
+    fun readPostgreCsvFile() {
+        writeFile("simple_postgre.csv", "id,name,balance\n1,Bob,10000.00")
+
+        val args = listOf("\"${dirPath("simple_postgre.csv")}\"", "{type:\"postgresql_csv\", header:true}").map { it.exprValue() }
+
+        val actual = function.call(env, args).ionValue
+        val expected = "[{id:\"1\",name:\"Bob\",balance:\"10000.00\"}]"
+
+        assertEquals(ion.singleValue(expected), actual)
+    }
+
+    @Test
+    fun readCustomizedCsvFile1() { // delimiter
+        writeFile("customized.csv", "id name balance\n1 Bob 10000.00")
+
+        val args = listOf("\"${dirPath("customized.csv")}\"", "{type:\"customized\", header:true, delimiter:' '}").map { it.exprValue() }
+
+        val actual = function.call(env, args).ionValue
+        val expected = "[{id:\"1\",name:\"Bob\",balance:\"10000.00\"}]"
+
+        assertEquals(ion.singleValue(expected), actual)
+    }
+
+    @Test
+    fun readCustomizedCsvFile2() { // ignore_empty_line
+        writeFile("customized.csv", "id,name,balance\n\n1,Bob,10000.00")
+
+        val args = listOf("\"${dirPath("customized.csv")}\"", "{type:\"customized\", header:true, ignore_empty_line: false}").map { it.exprValue() }
+
+        val actual = function.call(env, args).ionValue
+        val expected = "[{id:\"\"},{id:\"1\",name:\"Bob\",balance:\"10000.00\"}]"
+
+        assertEquals(ion.singleValue(expected), actual)
+    }
+
+    @Test
+    fun readCustomizedCsvFile3() { // trim and ignore_surrounding_space
+        writeFile("customized.csv", "id,name,balance\n 1 , Bob , 10000.00 ")
+
+        val args = listOf("\"${dirPath("customized.csv")}\"", "{type:\"customized\", header:true, ignore_surrounding_space:false, trim:false}").map { it.exprValue() }
+
+        val actual = function.call(env, args).ionValue
+        val expected = "[{id:\" 1 \",name:\" Bob \",balance:\" 10000.00 \"}]"
+
+        assertEquals(ion.singleValue(expected), actual)
+    }
```

Comment:
Not all the added custom configuration options have been tested here. Would be good to include them or create a separate issue to add them.

Similarly for the other csv `type`s like `mysql_csv`, `mongodb_csv`, etc.

Reply:
Will add tests for other custom configuration options in the next commit. 
However, the source of other csv types are a bit hard to get. For example, if we have to have `mysql_csv.txt` file, I think we need to actually use MySQL to generate it.